addons/crm_iap_lead_enrich/models/crm_lead.py

```diff
@@ -29,19 +30,21 @@ def _compute_show_enrich_button(self):
             else:
                 lead.show_enrich_button = True
 
-    @api.model
-    def _iap_enrich_leads_cron(self):
-        timeDelta = fields.datetime.now() - datetime.timedelta(hours=1)
-        # Get all leads not lost nor won (lost: active = False)
-        leads = self.search([
-            ('iap_enrich_done', '=', False),
-            ('reveal_id', '=', False),
-            ('probability', '<', 100),
-            ('create_date', '>', timeDelta)
-        ])
-        leads.iap_enrich(from_cron=True)
-
-    def iap_enrich(self, from_cron=False):
+    @api.model_create_multi
+    def create(self, vals_list):
+        leads = super(Lead, self).create(vals_list)
+        ConfigParameter = self.env['ir.config_parameter'].sudo()
+        enrich_mode = ConfigParameter.get_param('crm.iap.lead.enrich.setting', 'manual')
+        user_creation_mode = (not self._context.get('install_mode') or self._context.get('import_file')) and not getattr(threading.currentThread(), 'testing', False)
+        if enrich_mode == 'auto' and user_creation_mode:
+            enrich_limit = ConfigParameter.get_param('crm_iap_lead_encrich_limit', 1000)
```

Comment:
Still using encrich 8-) also please use dots to be coherent through all CRM config parameters use.